File: lib/core/services/gemini_service.dart
import 'package:google_generative_ai/google_generative_ai.dart';
import '../config/app_config.dart';

class GeminiService {
  late final ChatSession _chat;

  GeminiService() {
    final model = GenerativeModel(
      model: 'gemini-1.5-flash',
      apiKey: AppConfig.geminiApiKey,
      systemInstruction: Content.system(AppConfig.geminiSystemPrompt),
    );
    _chat = model.startChat();
  }

  Future<String> sendMessage(String text) async {
    final response = await _chat.sendMessage(Content.text(text));
    return response.text?.trim() ?? 'Sorry, I cannot answer right now.';
  }
}

File: src/core/services/geminiService.ts
import api from '../api/client';
import { getErrorMessage } from '../api/errors';

const HEALTH_INSIGHT_SESSION_ID = 'health-insight';

export class GeminiService {
  private sessionId: string = HEALTH_INSIGHT_SESSION_ID;

  async sendMessage(text: string): Promise<string> {
    try {
      const res = await api.post('/chat/message', {
        message: text,
        sessionId: this.sessionId,
      });

      const content: string | undefined = res.data?.content;
      if (!content || content.trim().length === 0) {
        return 'Sorry, I cannot answer right now.';
      }

      return content.trim();
    } catch (e) {
      console.warn(`Chat AI error: ${getErrorMessage(e)}`);
      return 'Sorry, I cannot answer right now.';
    }
  }

  reset(): void {
    this.sessionId = HEALTH_INSIGHT_SESSION_ID;
  }
}
